packages/blockchain-wallet-v4-frontend/src/store/index.js

```diff
@@ -42,6 +42,18 @@ const devToolsConfig = {
   serialize: serializer
 }
 
+const analyticsCaller = () => store => next => action => {
+  if (window.analytics) return next(action)
+
+  window.analytics = {
+    track(key, properties) {
+      store.dispatch(actions.analytics.trackEvent({ key, properties }))
+    }
+  }
```

Comment:
Well played Mr Dylan, well played

Reply:
Hah! Yeah so I needed a way to hijack `window.analytics` that also had access to dispatch so I could fire off our normal trackEvent actions. After a bunch of failed attempts at creating a new saga, I thought maybe a redux middleware could work and I think this is actually a really nice solution, albeit a hacky one.

Reply:
amazing one 🔥

Reply:
can you add comments explaining what you're assigning/overriding on window? 
this looks like a middleware. should it be in that folder?

Reply:
Moved this to it's own file and added comments.